### ISSUE TITLE
AttachOne and AttachMany eagerload in PostgreSQL: Varchar <> Integer comparison is not allowed.

### ISSUE THREAD

**Issue by RomainMazB** [octobercms/october#5200]
- OctoberCMS Build: ### 467
- PHP Version: 7.4
- Database Engine: PostgreSQL 12.3

### Description:
I'm facing an issue on RainLab\Blog with PostgreSQL which is mostly related to the `attachMany/attachOne` relationships: the `attachment_id` in the `system_files` table is stored as a `varchar(191)` (I guess to let developer use UUID on their models), but PostgreSQL can't process a string <> integer comparison in the `WHERE IN` clause.

So when you try to use the `blogPosts` component, which eager loads the `featured_images` from the category's posts, it fails:
> SQLSTATE[42883]: Undefined function: 7 ERREUR: l'opérateur n'existe pas : character varying = integer LINE 1: ...ystem_files" where "system_files"."attachment_id" in (1, 2) ... ^ HINT: Aucun opérateur ne correspond au nom donné et aux types d'arguments. Vous devez ajouter des conversions explicites de type. (SQL: select * from "system_files" where "system_files"."attachment_id" in (1, 2) and "system_files"."attachment_type" = RainLab\Blog\Models\Post and "field" = featured_images order by "sort_order" asc)

Loading a post with its featured image works properly in the blogPost component.

### Steps To Reproduce:
Fresh install on PostgreSQL with a model using attachMany relationship and then use eager load.

**Comment by bennothommo** [comment]
@RomainMazB does PostgreSQL have some form of "strict" mode like MySQL? If it does, perhaps disabling it will loosen up the type checking.

**Comment by RomainMazB** [comment]
Hi @bennothommo ,

AFAIK and after some searches to be sure of it: It doesn't exist a way to be less strict in PostgreSQL but I've found a fix and it's finally not related to PostgreSQL...

Hard to guess this issue has been added on Laravel 5.7!

On Laravel 5.7, the [whereInMethod](https://github.com/illuminate/database/blob/5.7/Eloquent/Relations/Relation.php#L317-L324) has been added to Eloquent. I didn't found where this method breaks the PostgreSQL query but it does: ensuring the `whereIn` method is used instead of `whereIntegerInRaw` method fixed the problem. I guess somewhere in the engine, a raw sql type cast is called (+ a strict comparison).

Since the `whereInMethod` function has been added on 5.7 ([it didn't exists in 5.6](https://github.com/illuminate/database/blob/5.6/Eloquent/Relations/Relation.php)). I've supposed this fix (or a better one) should be added on the Laravel 6 wip branch.

This is may be why this had not been noticed before.
